scipy/special/cephes.hh

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+/* Use this header to include functions from cephes in compiled special
+ * function kernels written in C++.
+
+ * cephes_names.h defines aliases airy -> cephes_airy etc.
+ * which causes trouble when one tries to use functions from cephes in the
+ * same translation unit where boost is used, due to name clashes. We undef
+ * all of these aliases and disambiguate the cephes functions by putting them
+ * in a scipy::special::cephes namespace.
+ */
+
+
+// Namespace the include to avoid polluting global namespace.
+namespace scipy {
+    namespace special {
+	namespace cephes {
+	    namespace cephes_internal {
```

Comment:
I would use `detail` here instead of `cephes_internal`, I think that's more standard practice. You can then have a `detail` in each namespace wherever you need it.